SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialEventTracer.cpp

```diff
@@ -127,6 +184,89 @@ void SpatialEventTracer::TraceEvent(const SpatialGDKEvent& Event)
 	}
 }
 
+void SpatialGDK::SpatialEventTracer::TraceEvent(EventName Name, EventType Type, const AActor* Actor, const UFunction* Function)
+{
+	Trace_SpanId CurrentSpanId = Trace_EventTracer_AddSpan(EventTracer, nullptr, 0);
+	Trace_Event TraceEvent{ CurrentSpanId, 0, TypeToString(Type), NameToString(Name), nullptr };
+	if (Trace_EventTracer_ShouldSampleEvent(EventTracer, &TraceEvent))
+	{
+		Trace_EventData* EventData = Trace_EventData_Create();
+
+		if (Actor != nullptr)
+		{
+
+			auto ActorName = StringCast<ANSICHAR>(*Actor->GetName());
+			const char* ActorNameStr = ActorName.Get();
+			Trace_EventData_AddStringFields(EventData, 1, &ActorKeyPtr, &ActorNameStr);
+
+			auto ActorPosition = StringCast<ANSICHAR>(*Actor->GetActorTransform().GetTranslation().ToString());
+			const char* ActorPositionStr = ActorPosition.Get();
+			Trace_EventData_AddStringFields(EventData, 1, &ActorPositionKeyPtr, &ActorPositionStr);
+
+		}
+
+		if (Function != nullptr)
+		{
+			auto FunctionName = StringCast<ANSICHAR>(*Function->GetName());
+			const char* FunctionNameStr = FunctionName.Get();
+			Trace_EventData_AddStringFields(EventData, 1, &FunctionKeyPtr, &FunctionNameStr);
+		}
+
+
+		TraceEvent.data = EventData;
+		Trace_EventTracer_AddEvent(EventTracer, &TraceEvent);
+		Trace_EventData_Destroy(EventData);
+	}
+}
+
+void SpatialGDK::SpatialEventTracer::TraceEvent(EventName Name, EventType Type, const AActor* Actor, ENetRole Role)
+{
+	Trace_SpanId CurrentSpanId = Trace_EventTracer_AddSpan(EventTracer, nullptr, 0);
+	Trace_Event TraceEvent{ CurrentSpanId, 0, TypeToString(Type), NameToString(Name), nullptr };
+	if (Trace_EventTracer_ShouldSampleEvent(EventTracer, &TraceEvent))
+	{
+		Trace_EventData* EventData = Trace_EventData_Create();
+
+		if (Actor != nullptr)
+		{
+			auto ActorName = StringCast<ANSICHAR>(*Actor->GetName());
```

Comment:
This code seems shared with the other TraceEvent. A function "AddStringFieldsForActor" could be abstracted.